### PR TITLE
Another round of annotation service refactoring/changes to make use of existing mappings in vidmapper.

### PR DESCRIPTION
**Pull request by nalinigans**
Also merge with develop. Will be working on annotation buffers next.

### CODE REVIEW

**Comment by kgururaj** [comment]
A general comment - looks like you're trying to build a vid mapper in C++ from the VCF headers. It struck me that it might be easier/better to do this in Python (using a library like cyvcf2 to parse the VCF headers) and create Protobuf vid structures:
* Large variation in VCF headers in terms of quality 
* Users can inspect json dumped and see if there're any issues
* Easier for bioinformaticians to tweak Python code if required

Not critical, food for thought

File: src/main/cpp/src/api/annotation_service.cc [hunk 1]
```diff
@@ -48,96 +100,40 @@ AnnotationService::AnnotationService(const std::string& export_configuration) {
...
   }
-  m_annotation_buffer.reserve(32);
+  m_annotation_buffer.reserve(64);
 }
 
 std::vector<annotation_source_t>& AnnotationService::get_annotation_sources() {
   return m_annotation_sources;
 }
 
+std::unordered_map<int, int> bcf_ht_type_to_nbytes = {
```
Comment (kgururaj) [inline src/main/cpp/src/api/annotation_service.cc:116]: VariantFieldTypeUtil::size(bcf_ht_type) gets you this  (variant_field_data.h)

Comment (nalinigans) [reply src/main/cpp/src/api/annotation_service.cc:116]: Ah, thanks!

**Comment by nalinigans** [comment]
> A general comment - looks like you're trying to build a vid mapper in C++ from the VCF headers. It struck me that it might be easier/better to do this in Python (using a library like cyvcf2 to parse the VCF headers) and create Protobuf vid structures:
> 
> * Large variation in VCF headers in terms of quality
> * Users can inspect json dumped and see if there're any issues
> * Easier for bioinformaticians to tweak Python code if required
> 
> Not critical, food for thought

Yep, this is interesting, let's discuss more on Friday. Is cyvcf2 better than htslib for parsing vcf headers? Or is it just that bioinformaticians will be able to tweak better in case of low quality VCF headers?

**Pull request merged**